app/Models/User.php

```diff
@@ -260,6 +263,16 @@ public function canAccessPanel(Panel $panel): bool
         return true;
     }
 
+    public function canImpersonate(): bool
+    {
+        return $this->can('authorization.impersonate');
+    }
+
+    public function canBeImpersonated(): bool
+    {
+        return $this->cannot('authorization.super_admin');
```

Comment:
This should probably just be a role check for `super_admin` I am not sure if we will be keeping this permission.